8.x/theme-development/helpers.md

```diff
@@ -1,100 +1,144 @@
+
+![create webpage in admin panel](/helpers/CreatePage.png)
+
+---
+
+### Helper functions description
+
+#### parse_url 
+
+*This is a php function which parses a URL and returns an associative array containing any of the various components of the URL that are present(such as `scheme`, `user`. `pass`, `host`, `port`,`path`, `query`, `fragment`). The values of the array elements are not URL decoded.* *<sup>[1](#1)</sup>*
+
+To do so, lets define a url and parse it:
 
 ```php
 @php
 
 $url= "https://google.com/salam/donya?test=sample";
-parse_url($url);
+dd(parse_url($url));
 
 @endphp
 ```
-Output
-``` 
+in `larammerce-base-theme` directory run:
+
+```bash
+./deploy.sh
+```
+
+Output:
+```bash
 array:4 [▼
   "scheme" => "https"
   "host" => "google.com"
   "path" => "/salam/donya"
   "query" => "test=sample"
 ]
 ```
-Each part of the URL that is ‍‍‍`scheme`, `host`, `path` and  `query` is displayed.
+As you see URL components are returned as an array.
 
+---
 
-Now an example is given to check the `unpars_url` function. Empty the `query` and unparse the URL.
+#### unparse_url
 
 ```php
-@php
+function unparse_url(array $parsed_url, array $ommit = []): string {
+    ...
+}
+```
+This function takes the parsed URL array as input, merges all the components, and as the result, returns a unified URL. 
+
+For instance, let's change the `query's` value and unparse the URL and see the result:
 
+```php{5,6}
+@php
+ 
 $url= "https://google.com/salam/donya?test=sample";
 $parsed_url = parse_url($url);
-
 $parsed_url["query"] = " ";
 $new_url = unparse_url($parsed_url);
 
 @endphp
 
 <p> the changed url : {{ $new_url }}</p>
 ```
-Output
-``` 
-"https://google.com/salam/donya? "
-```
-
-It can be seen that the query is removed from the URL.
+Output:
 
-This function takes an array of parsed URLs from the input and merges these into `scheme`, `user`, `pass`, `host`, `port`, `path`, `query`, `fragment` in the output and transforms them into a unified URL.
-___________
+```html
+<p> "https://google.com/salam/donya? " </p>
+```
 
+As you see, `test=sample` has been removed as an empty value is given to the query.
 
+---
 
+#### get_product_all_extras_percentage
+  
 
-### `get_product_all_extras_percentage`
 ```php
-function get_product_all_extras_percentage()
+ function get_product_all_extras_percentage() {
+        return \App\Utils\CMS\ProductService::getAllExtrasPercentage();
+    }
 ```
+This function returns the summation of `Tax percentage` and `Toll percentage` values.
 
-This function returns the sum of `getTaxpercentage` and `getToolpercentage` values.
-
+Example:
 ```php
 <p>Product all extra percentages  {{ get_product_all_extras_percentage() }}</p>
 ```
-Output
-``` 
-Product all extra percentages 9
-```
-_____________
+Output:
 
+```html
+<p> Product all extra percentages 9 </p>
```

Comment:
Delete p and /p tags in the output.

<p> Product all extra percentages 9 </p>  ->  Product all extra percentages: 9